src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadonlyStructTests.cs

```diff
@@ -1156,6 +1151,80 @@ .maxstack  2
 }");
         }
 
+        [Fact]
+        public void StackAllocSpanLengthConstantFolding()
+        {
+            var comp = CreateCompilationWithMscorlibAndSpan(@"
+using System;
+class Test
+{
+    public static void Main()
+    {
+        const int a = 5, b = 6;
+        Span<int> x = stackalloc int[a * b];
+        Console.Write(x.Length);
+    }
+}", TestOptions.ReleaseExe);
+
+            CompileAndVerify(comp, expectedOutput: "30", verify: Verification.Fails).VerifyIL("Test.Main", @"
+{
+  // Code size       26 (0x1a)
+  .maxstack  2
+  .locals init (System.Span<int> V_0) //x
+  IL_0000:  ldc.i4.s   120
+  IL_0002:  conv.u
+  IL_0003:  localloc
+  IL_0005:  ldc.i4.s   30
+  IL_0007:  newobj     ""System.Span<int>..ctor(void*, int)""
+  IL_000c:  stloc.0
+  IL_000d:  ldloca.s   V_0
+  IL_000f:  call       ""int System.Span<int>.Length.get""
+  IL_0014:  call       ""void System.Console.Write(int)""
+  IL_0019:  ret
+}");
+        }
+
+        [Fact]
+        public void StackAllocSpanLengthOverflow()
+        {
+            var comp = CreateCompilationWithMscorlibAndSpan(@"
+using System;
+class Test
+{
+    static void M()
+    {
+        Span<int> x = stackalloc int[int.MaxValue];
+    }
+
+    public static void Main()
+    {
+        try
+        {
+            M();
+        }
+        catch (OverflowException)
```

Comment:
I think you might not be able to catch this when it actually runs.

Reply:
Ah, no it is a regular overflow, not a StackOverflow.

Reply:
Actually this is the problem. On 64bit this is not an overflow since the native int can take this value, but then it becomes SO when we localloc.

I guess on 64bit we should not run the compiled code, only validate the IL.

Reply:
How can I repro locally? Should I add `ReleaseExe.WithPlatform(Platform.X86)`?

Reply:
or should I skip it for now?

Reply:
both `build` and `test` utilities in the repo root take `-test64` switch.

I think `build -test64` might do it.

Reply:
nope :-), just tried `build -test64` and it only did a build. Need to run `test -test64` after that.

Normally our test failures are not platform specific, so we just rely on CI

Reply:
I would just make a fix and push the change to the branch. 
It should really be something like:

```C#

var expectedIL = "....";


if (IntPtr.Size == 4)
{
       CompileAndVerify(comp, expectedOutput: "10", verify: Verification.Fails)
       . . . Verify (expectedIL):...
}
else
{
      // on 64bit the native int does not overflow, so we get StackOverflow instead
      // therefore we will just check the IL
      CompileAndVerify(comp, verify: Verification.Fails, options:TestOptions.ReleaseDll)
       . . . Verify (expectedIL):...
}
```